src/stores/OwnProfileStore.ts

```diff
@@ -136,12 +137,32 @@ export class OwnProfileStore extends AsyncStoreWithClient<IState> {
             if (!this.matrixClient) return;
             // We specifically do not use the User object we stored for profile info as it
             // could easily be wrong (such as per-room instead of global profile).
-            const profileInfo = await this.matrixClient.getProfileInfo(this.matrixClient.getSafeUserId());
+
+            let profileInfo: { displayname?: string; avatar_url?: string } = {
+                displayname: undefined,
+                avatar_url: undefined,
+            };
+
+            try {
+                profileInfo = await this.matrixClient.getProfileInfo(this.matrixClient.getSafeUserId());
+            } catch (error: unknown) {
+                if (!(error instanceof MatrixError) || error.errcode !== "M_NOT_FOUND") {
+                    /**
+                     * Raise any other error than M_NOT_FOUND.
+                     * M_NOT_FOUND could occur if there is no user profile.
+                     * {@link https://spec.matrix.org/v1.7/client-server-api/#get_matrixclientv3profileuserid}
+                     * We should then assume an empty profile, emit UPDATE_EVENT etc..
+                     */
+                    throw error;
+                }
```

Comment:
Maybe this would be nicer to have in `MatrixClient::getProfileInfo`?

Reply:
:thinking: makes sense…

Reply:
Or does it not make sense?

For our own profile we know it must exist (even if it is empty).

I can imagine some places that are going to explode if we globally switch from 404 to an empty profile, such as the invite dialog. However, the current implementation can also cause some strange problems because we rely on 404 in some cases to test whether a user exist.

It would be better if the endpoint only returns 404 if the user does not exist.

Reply:
MSC exists: https://github.com/matrix-org/matrix-spec-proposals/blob/travis/msc/profile-discovery/proposals/1797-profile-error-codes.md

Reply:
Good point